src/main/java/org/cactoos/io/TempFolder.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2017-2018 Yegor Bugayenko
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.cactoos.io;
+
+import java.io.Closeable;
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import org.cactoos.Scalar;
+import org.cactoos.Text;
+import org.cactoos.scalar.IoCheckedScalar;
+import org.cactoos.scalar.StickyScalar;
+import org.cactoos.text.JoinedText;
+import org.cactoos.text.RandomText;
+import org.cactoos.text.TextOf;
+
+/**
+ * A temporary folder.
+ * This is ephemeral folder to be used in small scopes.
+ * The physical folder is deleted from the filesystem when the temp folder is
+ * closed.
+ * @since 1.0
+ */
+public final class TempFolder implements Scalar<Path>, Closeable {
+
+    /**
+     * Creates the temporary folder, returning its path.
+     */
+    private final Scalar<Path> folder;
+
+    /**
+     * Ctor.
+     * Creates new folder in temporary directory
+     * with a random name.
+     * @since 1.0
+     */
+    public TempFolder() {
+        this(
+            new JoinedText(
+                new TextOf(""),
+                new TextOf("tmp"),
+                // @checkstyle MagicNumber (1 line)
+                new RandomText(5, 'a', 'b', 'c')
```

Comment:
@Iprogrammerr `RandomText` should cover `[a-zA-Z0-9]`.

Reply:
@fabriciofx How to do that? RandomText does not accept regex. You want me to create ListOf with 62 elements given in a ctor?

Reply:
@Iprogrammerr I said `[a-zA-Z0-9]` just to simplify. The idea is use `a...z`, `A...Z` and `0...9`. What's the problem with a ctor with 62 elements? Look inside `RamdomText`. It has a ctor with more than 62 elements (I just don't say to use this ctor because it has special characters too).